internal/api/arm/resource.go

```diff
@@ -149,3 +149,19 @@ func ListProvisioningStates() iter.Seq[ProvisioningState] {
 		ProvisioningStateUpdating,
 	})
 }
+
+type ExternalAuthProvisioningState string
```

Comment:
Hmm, I'm not sure if a separate string type is a good idea here.  The `database.OperationDocument` struct defines a `Status` field as `arm.ProvisioningState` and you'll need to use that for external auth operations.

For now I'd suggest just add an `"AwaitingSecret"` constant to the existing `ProvisioningState` type.